platform/linux-generic/arch/aarch64/odp/api/abi/atomic_inlines.h

```diff
@@ -152,7 +152,62 @@ static inline int _odp_atomic_cas_acq_rel_u128(odp_atomic_u128_t *atom, odp_u128
 
 	return 0;
 }
-#else /* _ODP_LOCK_FREE_128BIT_ATOMICS */
+
+static inline void _odp_atomic_add_u32(odp_atomic_u32_t *atom, uint32_t val)
+{
+	__asm__ volatile("stadd   %w0, [%1]"
+			 :
+			 : "r" (val), "r" (atom)
+			 : "memory");
+}
+
+static inline void _odp_atomic_sub_u32(odp_atomic_u32_t *atom, uint32_t val)
+{
+	__asm__ volatile("neg     %w0, %w0\n"
+			 "stadd   %w0, [%1]"
+			 : "+r" (val)
+			 : "r" (atom)
+			 : "memory");
```

Comment:
We would prefer avoiding the `neg` call if possible (seems to hinder inlining). Would the following code work?
```
	int32_t neg_val = -val;

	__asm__ volatile("stadd   %w0, [%1]"
			 :
			 : "r" (neg_val), "r" (atom)
			 : "memory");
```

Reply:
Hi Matias, I have changed the code to above in v4. It works as expected and all test cases pass with -O2 flag as well. Please review.